gapis/resolve/dependencygraph2/README.md

```diff
@@ -0,0 +1,165 @@
+capture for a given replay), but it might be relevant to other use-cases.
+
+## A dependency graph example
+
+The dependency graph is meant to be used internally and not directly presented
+to the end user. Still, in order to understand it, it is good to see what it
+looks like some simple examples.
+
+You can view the dependency graph of a given capture using the
+`create_graph_visualization` gapit verb, e.g.:
+
+```
+./gapit create_graph_visualization -format dot -out depgraph.dot com.example.gfxtrace
+```
+
+> Warning: the dependency graph gets quickly fairly big on traces of real games,
+> you might not be able to easily explore them using a DOT viewer.
+
+The dependency graph of a trace of one frame of Sascha Willems
+[triangle demo](https://github.com/SaschaWillems/Vulkan#01---triangle) looks
+like this (as of October 2020):
+
+<img alt="Dependency graph of a simple Vulkan triangle demo" src="triangle-depgraph.png" width="1000px"/>
+
+We can see that `vkQueuePresent` has a dependency edge pointing to
+`vkCmdEndRenderPass`, which reflects the fact that the render pass must be
+executed before the frame is presented. Now, `vkCmdEndRenderPass` itself
+transitively depends on all commands of the command buffer that was submitted,
+eventually leading to `vkQueueSubmit`. Also, `vkCmdDrawIndexed` depends on the
+several commands that performs all the required setup (`vkCmdBindPipeline`,
+`vkCmdBindIndexBuffer`, etc), but these setup commands do not depend on each
+other, as they can be performed in any order.
+
+## Structure of the dependency graph
+
+The `DependencyGraph` interface defines possible operations on a dependency
+graph, and it is implemented by the `dependencyGraph` type.
+
+The graph nodes are defined by the `Node` interface, which is implemented by two
+kinds of nodes: `CmdNode` represents an API command or subcommand, and `ObsNode`
+represents a memory observation. The graph also stores dependencies between the
+nodes: these dependencies are untyped, they are just represented by an ordered
+pair of node IDs, such that there is a source node and a target node.
+
+For all nodes, the graph stores the node accesses to API state and memory in a
+`NodeAccesses` type, which is retrievable via `GetNodeAccesses()`. Note that
+these accesses are saved only if the `SaveNodeAccesses` field of the
+`DependencyGraphConfig` object used at graph creation time is set to `true`.
+There are several kinds of node accesses, the main two are `FragmentAccess`
+which represent an access to a part (a "fragment") of the API state, and
+`MemoryAccess` which represents an access to some memory.
+
+## Creation of the dependency graph
+
+The dependency graph is created by mutating all the capture commands and
+monitoring their accesses to API state and memory.
+
+### Approach overview
+
+Let's give a high-level illustration of this process: consider we have a memory
+made of two slots. In each slot, we remember the ID of the node that was the
```

Comment:
Your first use of the word slot is to say "consider a memory made of two slots". Slot has not yet been defined.

Reply:
Clarified definition of slot in this context.